File: nameless_old/engine.js
// http://gamebanana.com/tuts/11229
/*
	This class is the main engine class. It contains all and does all! All hail the engine!
*/

var includes = ["character.js", "image.js", "npc.js", "object.js", "player.js", "socket.js", "sound.js", "sprite.js"];
var sprites = {};

function engine(){
	// Add include files
	for(var i = 0; i < includes.length; i++)
		loadScript(includes[i]);
	return this;
}

function loadScript(url){
	// Add scrip tag to head
	var head = document.getElementsByTagName("head")[0];
	var script = document.createElement("script");
	script.type = "text/javascript";
	script.src = url;

	// Fire the loading
	head.appendChild(script);
}

function addSprite(x_location, y_location, image_count, id, filename, width, height){
	sprites[id] = sprite(x_location, y_location, image_count);
	sprites[id].addImage(filename, 0, x_location, y_location, width, height);
}

function drawSprite(id, image_number){
	var canvas = document.getElementById("canvas");
	var context = canvas.getContext('2d');

	var img = new Image();
	img.src = sprites[id].images[image_number].getFilename();

	context.drawImage(img,
		sprites[id].images[image_number].getXLocation(),
		sprites[id].images[image_number].getYLocation(),
		sprites[id].images[image_number].getWidth(),
		sprites[id].images[image_number].getHeight(),
		sprites[id].getX(),
		sprites[id].getY(),
		sprites[id].images[image_number].getWidth(),
		sprites[id].images[image_number].getHeight(),
	);
}

engine();
addSprite(0,0,1,"circle", "test_image1.png", 16, 16);

File: nameless/Engine.java
import java.util.*;
import java.io.IOException;
/*
	This class is the main engine class. It contains all and does all! All hail the engine!
*/

public class Engine{

	private ArrayList<Sprite> sprites = new ArrayList<Sprite>();
	private boolean log_to_console;
	private Window window;
	private int refresh_rate;

	public Engine(String window_name, int x_location, int y_location, int width, int height){
		// Create game window
		window = new Window(window_name, x_location, y_location, width, height);
	}

	public void AddSprite(int x_location, int y_location, int image_count, int render_order, int id, String filename, int width, int height){
		sprites.add( new Sprite(x_location, y_location, image_count, render_order, filename) );
		sprites.get(id).AddImage(0, x_location, y_location, width, height);
	}

	public void AnimateSprite(int sprite_id){
		window.AnimateSprite(sprites.get(sprite_id));
	}

	public static void main(String[] args){
		Engine e1 = new Engine("Engine Tester", 10, 10, 800, 800);
		e1.AddSprite(0, 0, 1, 0, 0, "C:/Users/Josh/Documents/GitHub/simplicity/nameless/test_image1.png", 20, 20);
		e1.AnimateSprite(0);
	}

}